src/main/java/io/appium/java_client/AppiumClientConfig.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.java_client;
+
+import org.openqa.selenium.remote.http.ClientConfig;
+
+import java.time.Duration;
+
+/**
+ * Manage Appium Client configurations.
+ */
+
+public class AppiumClientConfig {
+    private boolean directConnect;
+
+    private ClientConfig clientConfig;
+
+    private static final Duration DEFAULT_READ_TIMEOUT = Duration.ofMinutes(10);
+
+    public AppiumClientConfig() {
+        this.clientConfig = ClientConfig.defaultConfig().readTimeout(DEFAULT_READ_TIMEOUT);
+        this.directConnect = false;
+    }
+
+    public static class DirectConnect {
+        public static String DIRECT_CONNECT_PROTOCOL = "directConnectProtocol";
+        public static String DIRECT_CONNECT_PATH = "directConnectPath";
+        public static String DIRECT_CONNECT_HOST = "directConnectHost";
+        public static String DIRECT_CONNECT_PORT = "directConnectPort";
+    }
+
+    /**
+     * Return the instance of AppiumClientConfig with a default config.
+     * @return the instance of AppiumClientConfig.
+     */
+    public static AppiumClientConfig defaultConfig() {
+        return new AppiumClientConfig();
+    }
+
+    /**
+     * Whether enable directConnect feature described in
+     * <a href="https://appiumpro.com/editions/86-connecting-directly-to-appium-hosts-in-distributed-environments">
+     *     Connecting Directly to Appium Hosts in Distributed Environments</a>.
+     *
+     * @param directConnect if enable the directConnect feature
+     * @return A self reference
+     */
+    public AppiumClientConfig directConnect(boolean directConnect) {
```

Comment:
lets use a common approach and have `set` prefix for all builder setter methods

Reply:
I wonder if here should follow `ClientConfig`, that had no `set`:

e.g.
```
  public ClientConfig baseUrl(URL baseUrl) {
    try {
      return baseUri(Require.nonNull("Base URL", baseUrl).toURI());
    } catch (URISyntaxException e) {
      throw new RuntimeException(e);
    }
  }

  public URI baseUri() {
    return baseUri;
  }
```

Still not decided yet to inherit, or just keep the clientconfig as a variable so that users can define their subclass of clientconfig as their clientconfig.